src/Public/LogRhythm/Case/General/Test-LrCaseIdFormat.ps1

```diff
@@ -22,41 +22,124 @@ Function Test-LrCaseIdFormat {
         IsGuid IsValid Value
         ------ ------- -----
          False    True 181
+    .EXAMPLE
+        C:\PS> Test-LrCaseIdFormat -Id "mock case"
+
+        LookupType : CaseName
+        IsValid    : False
+        Value      : mock case
+        CaseNumber :
+        CaseName   :
+        CaseGuid   :
+        Note       : Case Name value references more than one case.
     .LINK
         https://github.com/LogRhythm-Tools/LogRhythm.Tools
     #>
 
     [CmdletBinding()]
     Param(
-        [Parameter(
-            Mandatory = $true,
-            ValueFromPipeline = $true,
-            Position=0
-        )]
+        [Parameter(Mandatory = $true,ValueFromPipeline = $true,Position=0)]
         [ValidateNotNull()]
         [object] $Id
     )
 
-    $OutObject = [PSCustomObject]@{
-        IsGuid      =   $false
-        IsValid     =   $false
-        Value       =   $Id
+    begin {  
+        # https://docs.microsoft.com/en-us/dotnet/api/system.int32.tryparse
+        $_int = 0
     }
 
-    # https://docs.microsoft.com/en-us/dotnet/api/system.int32.tryparse
-    $_int = 0
-
-    # Check if ID value is an integer
-    if ([int]::TryParse($Id, [ref]$_int)) {
-        Write-Verbose "[$Me]: Id parses as integer."
-        $OutObject.Value = $Id.ToString()
-        $OutObject.IsValid = $true
-    # Check if ID value is a Guid
-    } elseif (($Id -Is [System.Guid]) -Or (Test-Guid $Id)) {
-        $OutObject.Value = $Id.ToString()
-        $OutObject.IsValid = $true
-        $OutObject.IsGuid = $true
+    process {
+        $OutObject = [PSCustomObject]@{
+            LookupType  =   $null
+            IsValid     =   $false
+            Value       =   $Id
+            CaseNumber  =   $null
+            CaseName    =   $null
+            CaseGuid    =   $null
+            Note        =   $null
+        }
+
+        # Check if ID value is an integer
+        if ([int]::TryParse($Id, [ref]$_int)) {
+            Write-Verbose "[$Me]: Id parses as integer."
+            $OutObject.Value = $Id.ToString()
+            $OutObject.LookupType = "CaseNumber"
+
+            # Lookup case by Number
+            try {
+                $Case = Get-LrCaseById -Id $Id
+                $OutObject.Note = "Case lookup performed by Case Number."
+            } catch {
+                $OutObject.IsValid = $false
+                $OutObject.Note = "Unable to retrieve results for Case Number lookup.  Number: $Id"
+            }
+
+            # Check result from Get-LrCaseById for presence of error object
+            if ($Case.error -eq $true) {
+                $OutObject.IsValid = $false
+                $OutObject.Note = $Case.Note
+            } else {
+                # Set output object results
```

Comment:
## Minor Bug

When lookup is by case number, nothing ever sets `IsValid` to `$true`. By default, `IsValid` is `$false`.  The only places that will set it to true occur in lookup by GUID (line 96) and when a name match is found (line 118).

Add `$OutObject.IsValid = $true` within the code block at line 78 should fix the issue.

I will continue my testing of this once commit is pushed - I am going through a few more tests before completed.

Reply:
Corrected the issue and placed the $OutObject.IsValid in a more logical location for all three test types.